src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -491,12 +492,14 @@ public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, HttpCompl
             CancellationTokenSource cts;
             bool disposeCts;
             bool hasTimeout = _timeout != s_infiniteTimeout;
+            long timeoutTime = long.MaxValue;
             if (hasTimeout || cancellationToken.CanBeCanceled)
             {
                 disposeCts = true;
                 cts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken, _pendingRequestsCts.Token);
                 if (hasTimeout)
                 {
+                    timeoutTime = Environment.TickCount64 + (long)_timeout.TotalMilliseconds;
```

Comment:
It probably doesn't matter, but you will get better perf if you avoid the long->double->long conversion here and use `(ulong)TotalTicks / (ulong)TicksPerMillisecond` rather than `(long)TotalMilliseconds`

Reply:
Done, but I didn't need casting here since all operands are long.